components/list/list-item.js

```diff
@@ -137,11 +180,17 @@ class ListItem extends RtlMixin(LitElement) {
 	}
 
 	render() {
-		const illustrationSlot = html`<slot name="illustration"></slot>`;
+		const contentId = getUniqueId();
+		const checkBoxId = getUniqueId();
+		const illustrationSlot = html`
+			<input @change="${this._handleChange}" class="d2l-list-item-checkbox" type="checkbox" id="${checkBoxId}">
```

Comment:
Perhaps we should only render the `input` if the item is selectable.  However that would require the items "know" whether they are selectable or not... and that would require either the list-item reaching out to check this on it's parent list component (not a huge fan of this), or adding a property to list-item.

Reply:
Yes, I considered both of those as well. All of them seem like they will have a performance hit. So I decided to go with the cleanest code approach for now.